src/ranged.cpp

```diff
@@ -1536,14 +1536,34 @@ void drop_or_embed_projectile( const dealt_projectile_attack &attack )
     item dropped_item = drop_item;
 
     monster *mon = dynamic_cast<monster *>( attack.hit_critter );
-    // Try to embed the projectile in monster
-    // Don't embed on miss, in player/NPCs, when we didn't stab/cut properly
-    //  or when the item simply shouldn't be embedded (for example, it is active)
-    if( mon == nullptr || mon->is_dead_state() ||
-        ( attack.dealt_dam.type_damage( DT_STAB ) +
-          attack.dealt_dam.type_damage( DT_CUT ) <=
-            attack.dealt_dam.type_damage( DT_BASH ) ) ||
-        effects.count( "NO_EMBED" ) != 0 ) {
+
+    // We can only embed in monsters
+    bool embed = mon != nullptr && !mon->is_dead_state();
+    // And if we actually want to embed
+    embed = embed && effects.count( "NO_EMBED" ) == 0;
+    // Don't embed in small creatures
+    if( embed ) {
+        const m_size critter_size = mon->get_size();
+        const int vol = dropped_item.volume( true, false );
```

Comment:
I might be reading it wrong, but the only reference to `unit_value` in `item::volume` I see is:
```
if ( unit_value == false ) {
    ret *= charges;
}
ret /= type->stack_size;
```
So for `unit_value == false` the volume should be of a single item of the stack rather than the whole stack.